pkg/broker/deprovision_subscriber.go

```diff
@@ -77,8 +82,10 @@ func cleanupDeprovision(instance *apb.ServiceInstance, dao *dao.Dao) error {
 	var err error
 	id := instance.ID.String()
 
-	if err = dao.DeleteExtractedCredentials(id); err != nil {
-		log.Error("failed to delete extracted credentials - %v", err)
+	err = dao.DeleteExtractedCredentials(id)
+	if err != nil && strings.Contains(err.Error(), etcdKeyNotFoundContents) {
```

Comment:
Same comment as the other PR, seems like we usually do `IsKeyNotFound` for etcd errors.

```
    if err != nil {
        // no jobs or states to recover, this is OK.
        if client.IsKeyNotFound(err) {
            log.Info("No jobs to recover")
            return "", nil
        }
        return "", err
    }
```

Reply:
@jmrodri +1. I'm unsure if we want to couple this subscriber etcd specifically, with an etcd `clientNotFound` check. It's probably fine because subscribers are designed to be specific to their use-cases (like persisting etcd data). What do you think?

Reply:
@jmrodri reading back over this, we definitely don't want a coupling point to etcd, especially with the move to CRDs. I think this needs to be it's own high level "ExtractedCredentialsNotFound" error. @shawn-hurley thoughts?

Reply:
I have to re-write this already because of the move to CRDs. 
I do think that having exposable DAO errors that our dao implementations uses makes sense here and something that is outside of the scope of this PR. 

It will be changed very soon, would rather be able to find all instances by searching for the `IsKeyNotFound` rather then having a one off.

Reply:
Cool, makes sense. I will update and ping you guys.